ocrd_models/ocrd_models/ocrd_page.py

```diff
@@ -61,6 +67,14 @@
 
 from .constants import NAMESPACES
 
+def di(ID):
```

Comment:
Generated code, monkey patch and then this - this PR clearly presents a path to the dark side...

More seriously, please convince me this is necessary yet safe enough to use here and not encounter side effects as also mentioned in the SO post comments. A very positive remark from the pilot libraries testing was that there didn't occur any crashes really.

Reply:
My goodness, this _is_ becoming serious! I agree with @cneud this might be a Pandorra's box and cost us stability and debuggability.

On the other hand, I am unqualified to suggest any alternatives for getting back to object references from etree nodes. This seems indeed to be the missing link. And `get_obj_by_id` does look like the ultimate prize for writing better processors.

Reply:
We could patch generateDS to keep actual references instead of memory adress srepresentations as the mapping key (i.e. replace the `id(obj)` calls with just `obj`). The intention with `id()` is to not prevent GC from freeing memory of references to generateDS classes that would be kept around as long as the mapping dict remains in scope. However, this is only in one direction, the reverse mapping uses etree Elements as key. And if anything changes in the PAGE via API, these references **should** be invalidated because the Elements do not reflect the state of the PAGE document anymore. If no dict is passed for the `mapping_` parameter, it uses a method-local dict to hold the references and this should not be a problem since the references would go out of scope immediately.

For the "search-by-id" and "iterate-over-all-region-types" operation, walking through the object tree every time nstead of using an index that can become invalidated all the time might be a worthy tradeoff. Essentially, reimplementing `ET.findall`.
 
In any case, monkey-patching is a bad idea, obviously, it's brittle and becomes invalid for every change to the document.